src/js/select2/results.js

```diff
@@ -466,7 +466,13 @@ define([
   Results.prototype.template = function (result, container) {
     var template = this.options.get('templateResult');
 
-    container.innerHTML = template(result);
+    var result = template(result);
```

Comment:
You're going to need to use a different variable here, we're clobbering `result`.